From c4aa290b36491976819ed5538d6a86e0dcb8ca9e Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:40:09 +0800
Subject: [PATCH 16/25] GSI Android 17 compatibility:
 0016-Use-a-fake-volume-policy-when-none-has-been-found.patch

Change-Id: I87371b938004ea6fa11ac6a0051522625beb52c9
---
 services/audiopolicy/engine/config/src/EngineConfig.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/services/audiopolicy/engine/config/src/EngineConfig.cpp b/services/audiopolicy/engine/config/src/EngineConfig.cpp
index 2aa57b4..a4f42e3 100644
--- a/services/audiopolicy/engine/config/src/EngineConfig.cpp
+++ b/services/audiopolicy/engine/config/src/EngineConfig.cpp
@@ -859,7 +859,12 @@ android::status_t parseLegacyVolumeFile(const char* path, VolumeGroups &volumeGr
 android::status_t parseLegacyVolumes(VolumeGroups &volumeGroups) {
     if (std::string audioPolicyXmlConfigFile = audio_get_audio_policy_config_file();
             !audioPolicyXmlConfigFile.empty()) {
-        return parseLegacyVolumeFile(audioPolicyXmlConfigFile.c_str(), volumeGroups);
+        int ret = parseLegacyVolumeFile(audioPolicyXmlConfigFile.c_str(), volumeGroups);
+        if (ret == NO_ERROR && volumeGroups.size() == 0) {
+            ret = parseLegacyVolumeFile("/system/etc/fake_audio_policy_volume.xml", volumeGroups);
+            ALOGE("Parsing volume for /system/etc/fake_audio_policy_volume.xml gave %zu",  volumeGroups.size());
+        }
+	return ret;
     } else {
         ALOGE("No readable audio policy config file found");
         return BAD_VALUE;
